src/networking.c

```diff
@@ -1651,11 +1610,7 @@ void clearClientConnectionState(client *c) {
     clientSetDefaultAuth(c);
     moduleNotifyUserChanged(c);
     discardTransaction(c);
-
-    pubsubUnsubscribeAllChannels(c, 0);
-    pubsubUnsubscribeShardAllChannels(c, 0);
-    pubsubUnsubscribeAllPatterns(c, 0);
-    unmarkClientAsPubSub(c);
+    freeClientPubSubData(c);
```

Comment:
So in this case we perform the actions of freeClient (eg all the dictReleases) when we unlink the client. Not sure it is problematic, but it is a change in order which is very basic.